src/components/ChatWindow/MessageList/MessageListScrollContainer/MessageListScrollContainer.tsx

```diff
@@ -0,0 +1,149 @@
+/* istanbul ignore file */
+import React from 'react';
+import ArrowDownwardIcon from '@material-ui/icons/ArrowDownward';
+import Button from '@material-ui/core/Button';
+import clsx from 'clsx';
+import { Message } from '@twilio/conversations/lib/message';
+import throttle from 'lodash.throttle';
+import { withStyles, WithStyles, createStyles } from '@material-ui/core/styles';
+
+const styles = createStyles({
+  messageListContainer: {
+    overflowY: 'auto',
+    flex: '1',
+  },
+  outerContainer: {
+    minHeight: 0,
+    flex: 1,
+    position: 'relative',
+  },
+  innerScrollContainer: {
+    height: '100%',
+    overflowY: 'auto',
+    padding: '0 1.2em 1em',
+  },
+  button: {
+    position: 'absolute',
+    bottom: '14px',
+    right: '2em',
+    zIndex: 100,
+    padding: '0.5em 0.9em',
+    visibility: 'hidden',
+    opacity: 0,
+    boxShadow: '0px 4px 16px rgba(18, 28, 45, 0.2)',
+    transition: 'all 0.5s ease',
+  },
+  showButton: {
+    visibility: 'visible',
+    opacity: 1,
+    bottom: '24px',
+  },
+});
+interface MessageListScrollContainerProps extends WithStyles<typeof styles> {
+  messages: Message[];
+}
+interface MessageListScrollContainerState {
+  isScrolledToBottom: boolean;
+  showButton: boolean;
+  messageNotificationCount: number;
+}
+
+/*
+ * This component is a scrollable container that wraps around the 'MessageList' component.
+ * The MessageList will ultimately grow taller than its container as it continues to receive
+ * new messages, and users will need to have the ability to scroll up and down the chat window.
+ * A "new message" button will be displayed when the user receives a new message, and is not scrolled
+ * to the bottom. This button will be hidden if the user clicks on it, or manually scrolls
+ * to the bottom. Otherwise, this component will auto-scroll to the bottom when a new message is
+ * received, and the user is already scrolled to the bottom.
+ *
+ * Note that this component is tested with Cypress only.
+ */
```

Comment:
@timmydoza do you mind proofreading this? want to make sure it's not **too** big

Reply:
Looks good! Not too big. Can't have too many comments 🙂